More specific messaging for unsupported block types

efischer19 opened the pull request:
A feature request in 2 pictures:

<img width="1196" alt="screen shot 2017-03-01 at 2 56 10 pm" src="https://cloud.githubusercontent.com/assets/7373924/23478420/531fac60-fe8f-11e6-979c-132dde9bbedb.png">
<img width="918" alt="screen shot 2017-03-01 at 2 56 34 pm" src="https://cloud.githubusercontent.com/assets/7373924/23478421/532006e2-fe8f-11e6-9ec2-4a531ba5487a.png">

Notice that despite only having a single unsupported feature in the list (edx-reverification-block), the messaging seems to indicate that you need to delete [every single unsupported component](https://github.com/edx/xblock-configuration/blob/master/xblock_configuration.json#L25). We have the ability to be much more specific here.

FYI @sstack22, who requested this. Turns out it was pretty easy to do!

I'm going to let Jenkins take a whack and see what fails, then add an additional test of my own before getting reviewers.

efischer19 commented:
All tests passed except a few python ones, which I've fixed and verified locally. Should be good to go now.

jcdyer commented on cms/djangoapps/contentstore/views/tests/test_course_index.py:
```diff
@@ -477,7 +477,7 @@ def _verify_deprecated_info(self, course_id, advanced_modules, info, deprecated_
         self.assertEqual(
             info['block_types_enabled'],
-            any(component in advanced_modules for component in deprecated_block_types)
+            [component for component in advanced_modules if component in deprecated_block_types]
```
This test just restates the logic of the code.  :(  

Would it make more sense to have a test that verifies that the components are all deprecated, and another test that verifies that we aren't showing components that the course isn't using?  That way the tests read more clearly as a statement of the behavior we're trying to acheive.

efischer19 replied:
> Would it make more sense to have a test that verifies that the components are all deprecated

It would, but that integration-level testing is far beyond the scope of this unit test IMO.

> another test that verifies that we aren't showing components that the course isn't using?

Added

jcdyer commented on cms/djangoapps/contentstore/views/course.py:
```diff
@@ -542,9 +542,9 @@ def _deprecated_blocks_info(course_module, deprecated_block_types):
     """
     data = {
         'block_types': deprecated_block_types,
```
I don't think we need to pass `block_types` to the template anymore.  If not, maybe we can rename `block_types_enabled` to `block_types` (for brevity with no loss of clarity).

efischer19 commented:
Need to make sure I'm using "unsupported" and "deprecated" accurately in all contexts

efischer19 commented:
I fixed up a method docstring that was out-of-date and replaced `unsupported_enabled_block_types` with `deprecated_enabled_block_types` for consistency while squashing commits.

efischer19 commented:
@jcdyer This is ready for another look
